src/lib/yang/alarms.lua

```diff
@@ -1,5 +1,53 @@
 module(..., package.seeall)
 
+local data = require('lib.yang.data')
+local util = require('lib.yang.util')
+
+local state = {
+   alarm_inventory = {},
+}
+
+function get_state ()
+   return {
+      alarm_inventory = state.alarm_inventory,
+   }
+end
+
+-- Single point to access alarm type keys.
+alarm_type_keys = {}
+
+function alarm_type_keys:fetch (...)
+   self.cache = self.cache or {}
+   local function lookup (alarm_type_id, alarm_type_qualifier)
+      if not self.cache[alarm_type_id] then
+         self.cache[alarm_type_id] = {}
+      end
+      return self.cache[alarm_type_id][alarm_type_qualifier]
+   end
+   local alarm_type_id, alarm_type_qualifier = unpack({...})
+   assert(alarm_type_id)
+   alarm_type_qualifier = alarm_type_qualifier or ''
+   local key = lookup(alarm_type_id, alarm_type_qualifier)
+   if not key then
+      key = {alarm_type_id=alarm_type_id, alarm_type_qualifier=alarm_type_qualifier}
+      self.cache[alarm_type_id][alarm_type_qualifier] = key
+   end
+   return key
+end
+
+local function load_default_configuration (filename)
+   filename = filename or "lib/yang/lwaftr-default-alarms.conf"
```

Comment:
Additionally I think nothing in `lib.yang` should be lwaftr-specific, if we can avoid it.